python/ray/air/_internal/session.py

```diff
@@ -74,6 +74,13 @@ def trial_resources(self) -> "PlacementGroupFactory":
         """Trial resources for the corresponding trial."""
         raise NotImplementedError
 
+    @property
+    def log_dir(self) -> str:
```

Comment:
Should this be `trial_dir`?

Reply:
I believe `log_dir` is more accurate here, since a Train session would return the worker-level log directory within the Tune trial directory. A Tune session would just return the trial dir.

Reply:
Tune and Train sessions shouldn't have diverging behaviors. How about we call this `trial_dir`, which always returns the trial dir, and then Train can have a separate call for worker-level log dirs?

Reply:
For the unified AIR session, should it be `session.get_trial_dir()` and `session.get_log_dir()`? For Tune sessions though, these two would return the same thing.

Reply:
Why don't we just not expose the log dir for now and only trial_dir? This is a new thing right? Train can always ignore the chdir flag anyways.

Actually, I don't think this PR should be having any considerations / impact for Train.

Reply:
Train currently also does a `chdir` to the worker-level log dir for checkpointing without contention between workers. I updated the PR to allow configuring Train to not do this chdir (basically having consistent functionality between Train and Tune).

Reply:
How about we implement that in a follow-up then? This PR can add only the Tune changes and only add an accessory for the tune trial dir. We can then add Train support.